specs/shared_inventory_openapi.yaml

```diff
@@ -21,48 +25,280 @@ paths:
+            minimum: 1
+            maximum: 100
+        - name: continue
+          in: query
+          description: The token for continuing to the next page of results
+          required: false
+          schema:
+            type: string
+      responses:
+        "200":
+          description: A paged list of kubernetes clusters registered in the current inventory.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/K8sClusterList'
+        "400":
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "401":
+          description: Unauthorized (really unauthenticated)
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "403":
+          description: Forbidden (really unauthorized)
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+    post:
+      summary: Create a K8s Cluster
+      operationId: createK8sCluster
+      requestBody:
+        description: The K8s Cluster to create
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/K8sCluster"
+      responses:
+        "200":
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/K8sCluster"
+        "400":
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "401":
+          description: Unauthorized (really unauthenticated)
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "403":
+          description: Forbidden (really unauthorized)
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  /api/inventory/v1.0/resources/k8s-clusters/{id}:
+    description: Manage a K8s Cluster
     parameters:
-      - name: resourceId
+      - name: id
         in: path
         schema:
           type: string
         required: true
         description: |-
           The resource ID assigned to the resource by the inventory.
-          A reporter alias may also be used as a resource ID using the format:
+           A reporter alias may also be used as a resource ID using the format:
```

Comment:
Could you elaborate how to generate the instanceId for the reporter? the same reporter may report multiple resources. how to ensure the instanceId is the same for the same reporter? do we store it somewhere?